### PR TITLE
Feat/join from mediadevice

### PR DESCRIPTION
**Pull request by leewardbound**
Solves #11 

BREAKING: necessarily includes this compatability fix https://github.com/pion/ion-sdk-go/pull/12 - i suggest merging #12 first.

Notice -- adding this example to this directory might inappropriately put extra dependencies (for webcam devices, etc) in go.mod -- is this appropriate?

### CODE REVIEW

File: example/join-from-mediadevice/main.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	for _, track := range s.GetTracks() {
+		track.OnEnded(func(err error) {
+			fmt.Printf("Track (ID: %s) ended with error: %v\n",
+				track.ID(), err)
+		})
+		_, err = c.Publish(track)
+		if err != nil {
+			panic(err)
+		} else {
+			break // only publish first track, thanks
```
Comment (leewardbound) [inline example/join-from-mediadevice/main.go:102]: I'm not sure why mediadevices would return 3 tracks, but I was seeing three of myself

File: go.sum [hunk 2]
```diff
@@ -622,6 +646,7 @@ gopkg.in/yaml.v2 v2.2.2/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.2.4/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.2.5/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.3.0/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
+gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c h1:dUUwHk2QECo/6vqA44rthZ8ie2QXMNeKRTHCNY2nXvo=
 gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
 honnef.co/go/tools v0.0.0-20180728063816-88497007e858/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
 honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
```
Comment (leewardbound) [inline go.sum:652]: @adwpc ^^ this is what i was referring to, is it appropriate to add all these extra deps in the `ion-sdk-go` repo? should we move this to `ion-examples`? i dont mind either way!

**Review by adwpc — APPROVED** [review]
LGTM!

File: go.sum (hunk 2 above)
Comment (adwpc) [reply go.sum:652]: Thanks, keep deps in this repo first

**Pull request merged**